### ISSUE TITLE
JSpecify: report unboxing warning for for-each loop over array of `@Nullable Integer`, etc.

### ISSUE THREAD

**Issue by msridhar** [uber/NullAway#1275]
```java
@NullMarked
class Test {
  void f(@Nullable Integer array) {
    for (int x : array) {}          // no warning right now
  }
}
```
In JSpecify mode we should give an unboxing warning for the above code.
